circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -445,4 +437,23 @@ public TableAndStatistics getTableAndStatistics(TableReplication tableReplicatio
         tableReplication.getReplicaTableName());
   }
 
+  public void checkIfReplicaCleanupRequired(
+      String replicaDatabaseName,
+      String replicaTableName,
+      DataManipulationClient dataManipulationClient) {
+    try (CloseableMetaStoreClient client = getMetaStoreClientSupplier().get()) {
+      if (replicationMode == FULL_OVERWRITE) {
+        LOG.debug("Replication mode: FULL_OVERWRITE. Checking for existing replica table.");
+        DropTableService dropTableService = new DropTableService();
+        try {
+            dropTableService
+              .dropTableAndData(client, replicaDatabaseName, replicaTableName,
+                  dataManipulationClient);
+        } catch (Exception e) {
+          LOG.info("Replica table '" + replicaDatabaseName + "." + replicaTableName + "' was not dropped.");
```

Comment:
Should we log `e` here too so we know why?

Reply:
should't we just throw the exception and stop?
We haven't deleted the existing one so writing again will probably fail anyway.